lib/iris/tests/unit/common/metadata/test_metadata_manager_factory.py

```diff
@@ -156,40 +156,38 @@ def setUp(self):
         kwargs = dict(zip(BaseMetadata._fields, values))
         self.metadata = metadata_manager_factory(BaseMetadata, **kwargs)
 
-    def test_pickle(self):
+    def test_pickle(self, tmp_path):
         for protocol in range(pickle.HIGHEST_PROTOCOL + 1):
-            with self.temp_filename(suffix=".pkl") as fname:
-                with open(fname, "wb") as fout:
-                    pickle.dump(self.metadata, fout, protocol=protocol)
-                with open(fname, "rb") as fin:
-                    metadata = pickle.load(fin)
-                    self.assertEqual(metadata, self.metadata)
+            fname = tmp_path / f"pickle_{protocol}.pkl"
+            with open(fname, "wb") as fout:
+                pickle.dump(self.metadata, fout, protocol=protocol)
+            with open(fname, "rb") as fin:
+                metadata = pickle.load(fin)
+                assert metadata == self.metadata
 
 
-class Test_instance__fields(tests.IrisTest):
-    def setUp(self):
+class Test_instance__fields:
+    @pytest.fixture(autouse=True)
+    def _setup(self):
         self.bases = BASES
 
     def test(self):
         for base in self.bases:
             fields = base._fields
             metadata = metadata_manager_factory(base)
-            self.assertEqual(metadata.fields, fields)
+            assert metadata.fields == fields
             for field in fields:
                 hasattr(metadata, field)
```

Comment:
Not a pytest issue, but I can't help being puzzled by this "hasattr" bit.
Is it a mistake, or does it merely test for "does not crash" ?

Reply:
I guess it's a pattern that I used to ensure that the fields are genuinely accessible.

We could add an `assert` here

Reply:
See lines +62 and +64 ... yup, you've spotted a genuine omission 👍